clients/drcachesim/tests/raw2trace_io.cpp

```diff
@@ -157,10 +155,57 @@ main(int argc, const char *argv[])
         std::string error = raw2trace.do_conversion();
         if (!error.empty()) {
             std::cerr << "raw2trace failed " << error << "\n";
-            return 1;
+            return false;
         }
         std::cerr << "Processed\n";
-        return 0;
+        exit(0);
+        // Unreachable
+        return false;
     }
-    return 0;
+}
+
+bool
+test_module_mapper(raw2trace_directory_t *dir)
+{
+    module_mapper_t mapper(dir->modfile_bytes);
+    EXPECT(mapper.get_last_error().empty(), "Module mapper construction failed");
+    REPORT("About to load modules");
+    const auto &loaded_modules = mapper.get_loaded_modules();
+    EXPECT(!loaded_modules.empty(), "Expected module entries");
+    EXPECT(mapper.get_last_error().empty(), "Module loading failed");
+    REPORT("Loaded modules successfully");
+    bool found_simple_app = false;
+    for (const module_t &m : mapper.get_loaded_modules()) {
+        std::string path = m.path;
+        if (path.rfind("simple_app") != std::string::npos) {
+            found_simple_app = true;
+            break;
+        }
+    }
+    EXPECT(found_simple_app, "Expected app entry not found in module map");
+    REPORT("Successfully found app entry");
+    return true;
+}
+
+int
+main(int argc, const char *argv[])
+{
+    std::string parse_err;
+    if (!droption_parser_t::parse_argv(DROPTION_SCOPE_FRONTEND, argc, (const char **)argv,
+                                       &parse_err, NULL) ||
+        op_indir.get_value().empty() || op_out.get_value().empty()) {
+        std::cerr << "Usage error: " << parse_err << "\nUsage:\n"
+                  << droption_parser_t::usage_short(DROPTION_SCOPE_ALL);
+        return 1;
+    }
+
+    /* Open input/output files outside of traced region. And explicitly don't destroy dir,
+     * so they never get closed.
+     */
+    raw2trace_directory_t *dir =
+        new raw2trace_directory_t(op_indir.get_value(), op_out.get_value());
+
+    bool test1_ret = test_raw2trace(dir);
+    bool test2_ret = test_module_mapper(dir);
+    return !(test1_ret && test2_ret);
```

Comment:
style: prefer keeping types separate (i.e., bools aren't ints) -- maybe sthg like this:

```C
if (!test_raw2trace(dir))
    return 1;
if (!test_module_mapper(dir))
    return 1;
return 0;
```

Reply:
Done, but kept the behavior where we can run the second test even if the first one fails, and then fail. This way, if both were to fail, the output and the diff to the output would reflect that.